WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -48,11 +55,17 @@ public void onCreate(Bundle savedInstanceState) {
         } else {
             mSite = (SiteModel) savedInstanceState.getSerializable(WordPress.SITE);
         }
+
+        if (mSite == null) {
+            mSite = mSiteStore.getSiteByLocalId(AppPrefs.getSelectedSite());
```

Comment:
Updated to pass the model in the intent in 59672d8.

I'll also experiment with sending the user directly to the editor.

(As we discussed, we might also want to split the behavior of tapping on success vs failure notifications, though not necessarily in this PR.)